### PR TITLE
Add support for writing Ext TCP Hdr TLV

### PR DESCRIPTION
**Pull request by GregoryVds**
Add support for writing an empty Extended TCP Header TLV.
We would need to add C unit tests at some point, as this case is not covered by the client library.

### CODE REVIEW

File: convert_util.h [hunk 1]
```diff
@@ -71,6 +71,10 @@ struct convert_opts {
 	/* if CONVERT_F_ERROR is set in flags */
 	uint8_t		error_code;
 
+	/* if CONVERT_F_EXTENDED_TCP_HDR is set in flags
```
Comment (gdetal) [inline convert_util.h]: You could do something like:

```
uint8_t *options;
size_t n_options;
```

options needs to be malloc'ed and free'd by the caller.

File: convert_util.c [hunk 2]
```diff
@@ -162,6 +162,23 @@ _convert_write_tlv_error(uint8_t *buff, size_t buff_len,
 	return length;
 }
 
+static ssize_t
+_convert_write_tlv_extended_tcp_hdr(uint8_t *buff, size_t buff_len,
+                                    UNUSED struct convert_opts *opts)
```
Comment (gdetal) [inline convert_util.c]: remove UNUSED

File: convert_util.c [hunk 3]
```diff
@@ -162,6 +162,23 @@ _convert_write_tlv_error(uint8_t *buff, size_t buff_len,
...
+{
+	struct convert_extended_tcp_hdr *ext_tcp_hdr =
+		(struct convert_extended_tcp_hdr *)buff;
+	size_t length = CONVERT_ALIGN(
+		sizeof(*ext_tcp_hdr));
+
+	if (buff_len < length)
+		return -1;
+
+	/* TODO support actually announcing TCP options */
```
Comment (gdetal) [inline convert_util.c]: Here you can go through the opts->options and write them in the TLV.

**Review by gdetal — CHANGES_REQUESTED** [review]
Thanks for the change. Can you update the code such that the Convert Header TLV can be filled with TCP options ?

File: convert_util.h (hunk 1 above)
Comment (GregoryVds) [reply convert_util.h]: Done for writing. Will add support for reading in another commit.

File: convert_util.c (hunk 2 above)
Comment (GregoryVds) [reply convert_util.c]: Done

File: convert_util.c (hunk 3 above)
Comment (GregoryVds) [reply convert_util.c]: Done

**Review by gdetal — APPROVED** [review]

**Pull request merged**